### PR TITLE
pythonPackages.urwid: 1.3.1 -> 2.0.1

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

A new package I'd like to contribute depends on urwid>=2.0.0, from upstream's [changelog](http://urwid.org/changelog.html), it seems no breaking changes were done. I hope this is reliable enough and that this update won't break existing packages.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/urwid/default.nix [hunk 1]
```diff
@@ -2,31 +2,13 @@
...
-   # fix tests
-   (fetchpatch {
-     url = "https://github.com/floppym/urwid/commit/f68f2cf089cfd5ec45863baf59a91d5aeb0cf5c3.patch";
-     sha256 = "1b3vz7mrwz2bqvdwvbyv2j835f9lzapgw0j2km4sam76bxmgfpgq";
-   })
-  ];
-
-  postPatch = ''
-    # Several tests keep failing on Hydra
-    rm urwid/tests/test_vterm.py
```
Comment (worldofpeace) [inline pkgs/development/python-modules/urwid/default.nix]: Are we certain the test still won't fail?
* https://github.com/urwid/urwid/blob/master/urwid/tests/test_vterm.py

File: pkgs/development/python-modules/urwid/default.nix [hunk 2]
```diff
@@ -2,31 +2,13 @@
 
 buildPythonPackage (rec {
```
Comment (worldofpeace) [inline pkgs/development/python-modules/urwid/default.nix]:
```suggestion
buildPythonPackage rec {
```

the one at the end of the file as well

File: pkgs/development/python-modules/urwid/default.nix [hunk 3]
```diff
@@ -2,31 +2,13 @@
```
Comment (worldofpeace) [inline pkgs/development/python-modules/urwid/default.nix]: Drop `fetchpatch` here

File: pkgs/development/python-modules/urwid/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/development/python-modules/urwid/default.nix]: Done

File: pkgs/development/python-modules/urwid/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/development/python-modules/urwid/default.nix]: I've had it running tests on my machine and nothing failed, everything ran smooth. If a test will fail on hydra and not on my machine, I can never know.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build pythonPackages.urwid python35Packages.urwid python36Packages.urwid python37Packages.urwid

File: pkgs/development/python-modules/urwid/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/urwid/default.nix]: it is a major bump, they could have fixed some of their flakey tests

**Review by worldofpeace — APPROVED** [review]
Builds fine on ofborg :+1:

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS (1 package broken on master)
diff LGTM

```
[26 built (1 failed)]
error: build of '/nix/store/i2dnj3513448k8ibh3nixzy519szcpsv-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/67096
1 package failed to build:
python37Packages.python_openzwave

27 package were build:
alot khal mitmproxy python27Packages.bpython python27Packages.configshell python27Packages.pudb python27Packages.pycarddav python27Packages.upass python27Packages.urwid python27Packages.urwidtrees python37Packages.bpython python37Packages.configshell python37Packages.homeassistant-pyozw python37Packages.pudb python37Packages.upass python37Packages.urwid python37Packages.urwidtrees s-tui salut_a_toi speedometer targetcli todoman topydo turses urlscan wicd wikicurses
```

**Comment by worldofpeace** [comment]
@doronbehar Can you squash commits?

**Comment by doronbehar** [comment]
Squashed.

**Comment by worldofpeace** [comment]
Thanks 🎆️

**Pull request merged**